db/links/operations/create.py

```diff
@@ -0,0 +1,67 @@
+from alembic.operations import Operations
+from alembic.migration import MigrationContext
+from sqlalchemy import MetaData
+
+from db.columns.base import MathesarColumn
+from db.constraints.utils import naming_convention
+from db.tables.operations.select import reflect_table_from_oid, reflect_tables_from_oids
+from db.tables.utils import get_primary_key_column
+
+
+def create_foreign_key_link(engine, schema, column_name, referrer_table_oid, referent_table_oid, unique_link=False):
+    with engine.begin() as conn:
+        referent_table = reflect_table_from_oid(referent_table_oid, engine, conn)
+        referrer_table = reflect_table_from_oid(referrer_table_oid, engine, conn)
+        primary_key_column = get_primary_key_column(referent_table)
+        metadata = MetaData(bind=engine, schema=schema, naming_convention=naming_convention)
+        opts = {
+            'target_metadata': metadata
+        }
+        ctx = MigrationContext.configure(conn, opts=opts)
+        op = Operations(ctx)
+        column = MathesarColumn(
+            column_name, primary_key_column.type
+        )
+        op.add_column(referrer_table.name, column, schema=schema)
+        if unique_link:
+            op.create_unique_constraint(None, referrer_table.name, [column_name], schema=schema)
+        op.create_foreign_key(
+            None,
+            referrer_table.name,
+            referent_table.name,
+            [column.name],
+            [primary_key_column.name],
+            source_schema=schema,
+            referent_schema=schema
+        )
```

Comment:
Unless there's a convincing reason not to, I think you should create the column with the unique constraint and foreign key included, then add that column to the table. This would make this simpler and more aligned with the foreign key creation in `tb/tables/operations/split.py`. I can think of no reason to do it in separate operations given that we're not supporting multi-column foreign key creation with this.

Reply:
Isn't creating a foreign key constraint in individual steps much more flexible, as it could allow for multiple columns, although we don't use it now. Does creating a Foreign key as part of the column bring in any advantage?

Reply:
I think the main advantage is that it's easier to see the logic. I acknowledge, however, that it's more flexible in separate steps. If you want to keep separate steps, I can go with it. Consider using your already-written `create_foreign_key_link` function to add the foreign key.